subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/IsolatedAntBuilderMemoryLeakIntegrationTest.groovy

```diff
@@ -119,17 +137,18 @@ class IsolatedAntBuilderMemoryLeakIntegrationTest extends AbstractIntegrationSpe
         ]
     }
 
+    @Requires(TestPrecondition.JDK11_OR_LATER) // grgit 5 requires JDK 11, see https://github.com/ajoberstar/grgit/issues/355
     void "does not fail with a PermGen space error or a missing method exception"() {
         given:
         initGitDir()
         buildFile << """
             buildscript {
               repositories {
-                ${gradlePluginRepositoryDefinition()}
+                ${mavenCentralRepository()}
               }
 
               dependencies {
-                classpath "org.ajoberstar.grgit:grgit-gradle:4.1.1"
+                classpath "org.ajoberstar.grgit:grgit-core:5.0.0"
               }
             }
```

Comment:
I should have noted - grgit 4.x cannot run with Groovy 4.  So by bumping the grgit version to 5.x here, it runs with both Groovy 3 and 4 - at the expense of requiring JDK 11.